src/modality-agnostic-files/provenance.md

```diff
@@ -0,0 +1,487 @@
+    ![](../images/provenance_definitions_graph.png)
+
+**Provenance group**: refers to a collection of provenance records corresponding to a consistent set of processings. Defining multiple provenance groups is appropriate when separate sets of processings have been performed on data, and that they differ in purpose or chronology.
+
+## Provenance in sidecar JSON files
+
+Provenance metadata MAY be stored inside the sidecar JSON of any BIDS file it applies to.
+This metadata only describes the provenance of the BIDS file.
+
+!!! example
+    Provenance metadata in a sidecar JSON file:
+    ```JSON
+    {
+        "GeneratedBy": "bids::prov#conversion-00f3a18f",
+        "SidecarGeneratedBy": [
+            "bids::prov#preparation-conversion-1xkhm1ft",
+            "bids::prov#conversion-00f3a18f"
+        ],
+        "Digest": {
+            "sha256": "66eeafb465559148e0222d4079558a8354eb09b9efabcc47cd5b8af6eed51907"
+        }
+    }
+    ```
+    This snippet is derived from the following comprehensive example: [Provenance records for DICOM to Nifti conversion using `heudiconv`](https://github.com/bclenet/bids-examples/tree/BEP028_heudiconv/provenance_heudiconv).
+
+Any sidecar JSON file MAY include the following keys:
+
+<!-- This block generates a metadata table.
+The definitions of these fields can be found in
+  src/schema/objects/metadata.yaml
+and a guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_metadata_table(
+   {
+      "GeneratedById": "OPTIONAL",
+      "SidecarGeneratedBy": "OPTIONAL",
+      "Digest": "OPTIONAL",
+      "EntityType": "OPTIONAL"
+   }
+) }}
+
+!!! note
+    Not defining the `SidecarGeneratedBy` field means that the sidecar JSON was generated by the same `Activity` as the data file itself and described in the `GenearatedBy` field.
+
+    `SidecarGeneratedBy` MUST explicitly include all activities that modified the sidecar JSON, even if some are already listed in `GeneratedBy`.
+
+## Provenance at dataset level
+
+Provenance metadata MAY be stored inside the `dataset_description.json` of any BIDS dataset (or BIDS-Derivatives dataset) it applies to.
+This metadata describes the provenance of the whole dataset.
+
+!!! example
+    `GeneratedBy` contents in a `dataset_description.json`:
+    ```JSON
+    {
+        "GeneratedBy": [
+            {
+                "Name": "fmriprep",
+                "Id": "bids::prov#preprocessing-xMpFqB5q"
+            }
+        ]
+    }
+    ```
+    This snippet is an extract of the following comprehensive example: [Provenance records for fMRI preprocessing using `fMRIPrep`](https://github.com/bclenet/bids-examples/tree/BEP028_fmriprep/provenance_fmriprep).
```

Comment:
if we are going to point to a separate repo, we should note that it will have to be maintained after the PR is merged. perhaps something to think about and consider whether these examples are transitioned into bids examples repo and linked here.